components/tikv_kv/src/lib.rs

```diff
@@ -234,7 +234,9 @@ pub trait Engine: Send + Clone + 'static {
 /// at a specific timestamp. This snapshot is lower-level, a view of the underlying storage.
 pub trait Snapshot: Sync + Send + Clone {
     type Iter: Iterator;
-    type Ext<'a>: SnapshotExt;
+    type Ext<'a>: SnapshotExt
+    where
+        Self: 'a;
```

Comment:
Is it from a clippy warning? I don't see clippy complaining without the where clause.

Are you running clippy with a more recent toolchain?

Reply:
Yes, I ran cargo fix with nightly-2021-12-17, it seems more robust than our toolchain.

Reply:
OK, I see this is due to this GAT change https://github.com/rust-lang/rust/issues/87479